Source/WebAssembly/WebAssemblyTextPrint.cpp

```diff
@@ -11,6 +11,22 @@
 
 using namespace AST;
 
+bool startsWith(const char *str, const char *prefix)
+{
+        size_t lenstr = strlen(str);
+        size_t lenprefix = strlen(prefix);
+        if(lenstr < lenprefix) { return false; }
+	return (strncmp(prefix, str, lenprefix) == 0);
+}
+
+bool nameConflict(const char *name)
+{
+	return (startsWith(name, "local") ||
+		startsWith(name, "func") ||
+		startsWith(name, "sig") ||
+		startsWith(name, "label"));
```

Comment:
Could also change the generate names to all start with _ to further decrease the odds of a hit